src/Rector/Property/DoctrineTargetEntityStringToClassConstantRector.php

```diff
@@ -0,0 +1,203 @@
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\BetterPhpDocParser\PhpDocParser\ClassAnnotationMatcher;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\Doctrine\NodeAnalyzer\AttributeFinder;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\Rector\Property\DoctrineTargetEntityStringToClassConstantRector\DoctrineTargetEntityStringToClassConstantRectorTest
+ */
+final class DoctrineTargetEntityStringToClassConstantRector extends AbstractRector implements MinPhpVersionInterface
+{
+    private const ATTRIBUTE_NAME__TARGET_ENTITY = 'targetEntity';
+
+    private const ATTRIBUTE_NAME__CLASS = 'class';
+
+    /**
+     * @var array<class-string<OneToMany|ManyToOne|OneToOne|ManyToMany|Embedded>, string>
+     */
+    private const VALID_DOCTRINE_CLASSES = [
+        OneToMany::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        ManyToOne::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        OneToOne::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        ManyToMany::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        Embedded::class => self::ATTRIBUTE_NAME__CLASS,
+    ];
+
+    public function __construct(
+        private readonly ClassAnnotationMatcher $classAnnotationMatcher,
+        private readonly AttributeFinder $attributeFinder
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Convert targetEntities defined as String to <class>::class Constants in Doctrine Entities.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /**
+     * @ORM\OneToMany(targetEntity="AnotherClass")
+     */
+    private readonly ?Collection $items;
+
+    #[ORM\ManyToOne(targetEntity: "AnotherClass")]
+    private readonly ?Collection $items2;
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /**
+     * @ORM\OneToMany(targetEntity=\Rector\Doctrine\Tests\Rector\Property\DoctrineTargetEntityStringToClassConstantRector\Source\AnotherClass::class)
+     */
+    private readonly ?Collection $items;
+
+    #[ORM\ManyToOne(targetEntity: \Rector\Doctrine\Tests\Rector\Property\DoctrineTargetEntityStringToClassConstantRector\Source\AnotherClass::class)]
+    private readonly ?Collection $items2;
+}
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        // The minimum Version is PHP 5.5, because we need classname constants,
+        // and support Annotations as well as Attributes.
+//        return PhpVersionFeature::ATTRIBUTES;
+        return PhpVersionFeature::CLASSNAME_CONSTANT;
+    }
```

Comment:
This method is needed only for pure-PHP related rules.
This is related with Doctrine. So it should be checked in `refactor()` method. But this PHP version is so old it can be safely skipped.